### ISSUE TITLE
[Feature Request] Open instance of Insomnia in read-only data

### ISSUE THREAD

**Issue by gnud** [getinsomnia/insomnia#1763]
Imagine you want to copy or compare data from another workspace and it happens you have like several screens.

I propose few options:

1. Open new window from the same instance from app menu like open new window
2. Open new instance in read-only via command line arguments
3. Hybrid version, in Switch Workplace menu, a link button that opens in read-only window something like Incognito

A Spy logo is showing up next to the Workspace name or something when data is opened with read-only.

Can this be developed?
I can help, but I would need some assistance.

Any takers :)?

Thanks.

**Comment by gschier** [comment]
This sounds like it could be solved by #667? Does it have to be read-only?

**Comment by gnud** [comment]
> This sounds like it could be solved by #667? Does it have to be read-only?
If the feature can be done as concurent app instance + parallel read/write to the database, then no. 

Lets see some use cases, having two monitors a users wants to compare old vs new versions of the API.
At the moment we could have an infinite number of app instances as far as I tested.
Every instance can edit the data, but the last app instance closed overwrites the data.

The desired behaviour is to have some sort of instance checker on startup if the app has a lock then show a message "app is already running, should we open in readonly mode?"

What do you say about the idea?

**Comment by gschier** [comment]
I think having multi-window support would have a higher effort/value ratio so that's the direction I would lean.

If you are able to open the application multiple times, it's a bug. The app is currently supposed to prevent that from happening because, like you said, the database does not support concurrent access.

**Comment by gnud** [comment]
Yes, I am all for the mutl-window support.

At the moment I am able to open the app in many instances which leads to coruption of the work saved in the previos app instance.
Will this work:
https://github.com/electron/electron/blob/v0.36.10/docs/api/app.md#appmakesingleinstancecallback
?

**Comment by gschier** [comment]
Yep, exactly, which is weird because Insomnia already does that https://github.com/Kong/insomnia/blob/0a585395ec985d38c626c3b88b4cec1a69e9454f/packages/insomnia-app/app/main.development.js#L83
